### PR TITLE
feat(query): enable empty projection

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

1. If there is empty projection, we just return an empty block with num_rows.
2. Reduce extra stats clone in `pruning_blocks`

Closes #issue

### CODE REVIEW

**Review by leiysky — APPROVED** [review]

File: src/query/expression/src/kernels/scatter.rs [hunk 1]
```diff
@@ -41,6 +41,15 @@ use crate::Value;
 impl DataBlock {
     pub fn scatter<I>(&self, indices: &[I], scatter_size: usize) -> Result<Vec<Self>>
     where I: common_arrow::arrow::types::Index {
+        if indices.is_empty() || self.num_columns() == 0 {
+            let mut result = Vec::with_capacity(scatter_size);
+            result.push(self.clone());
+            for _ in 1..scatter_size {
+                result.push(self.slice(0..0));
```
Comment (RinChanNOWWW) [inline src/query/expression/src/kernels/scatter.rs:48]: The num_rows of each scattered block should according to the indices.

For example: 

indices: [0, 0, 1, 2, 1], scatter: 3.

The scattered blocks' num_rows are:

[2, 2, 1]

Comment (sundy-li) [reply src/query/expression/src/kernels/scatter.rs:48]: But the column or the slice is empty here

File: src/query/sql/src/planner/optimizer/rule/rewrite/rule_push_down_limit_sort.rs [hunk 2]
```diff
@@ -73,7 +73,7 @@ impl Rule for RulePushDownLimitSort {
             let mut sort_limit: logsort = sort.plan().clone().try_into()?;
             sort_limit.limit = Some(sort_limit.limit.map_or(count, |c| cmp::max(c, count)));
             let sort = SExpr::create_unary(RelOperator::Sort(sort_limit), sort.child(0)?.clone());
-            state.add_result(s_expr.replace_children(vec![sort]));
```
Comment (RinChanNOWWW) [inline src/query/sql/src/planner/optimizer/rule/rewrite/rule_push_down_limit_sort.rs:76]: What is the difference of this modification?

File: src/query/sql/src/planner/plans/join.rs [hunk 3]
```diff
@@ -386,8 +386,10 @@ impl Operator for Join {
         let probe_physical_prop = rel_expr.derive_physical_prop_child(0)?;
         let build_physical_prop = rel_expr.derive_physical_prop_child(1)?;
 
+        // it join/probe side is Serial or join key is empty, we use Serial distribution
```
Comment (RinChanNOWWW) [inline src/query/sql/src/planner/plans/join.rs:389]: it -> if?

File: src/query/service/src/api/rpc/exchange/exchange_transform.rs [hunk 4]
```diff
@@ -329,18 +330,25 @@ impl ExchangeTransform {
             &Default::default(),
         )?;
 
-        let meta = match bincode::deserialize(fragment_data.get_meta()) {
+        let meta = match bincode::deserialize(&fragment_data.get_meta()[4..]) {
```
Comment (RinChanNOWWW) [inline src/query/service/src/api/rpc/exchange/exchange_transform.rs:333]: How about using a CONST to represent this `4` and add some comments.

**Comment by RinChanNOWWW** [comment]
I'm curious about why results of some explain tests changed.

**Review by RinChanNOWWW — APPROVED** [review]

**Pull request merged**